packages/fast-components-react-msft/src/action-trigger/action-trigger.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { Button, ButtonAppearance } from "../button";
+import {
+    ActionTriggerAppearance,
+    ActionTriggerHandledProps,
+    ActionTriggerUnhandledProps,
+} from "./action-trigger.props";
+import { actionTriggerButtonOverrides } from "@microsoft/fast-components-styles-msft";
+
+class ActionTrigger extends Foundation<
+    ActionTriggerHandledProps,
+    ActionTriggerUnhandledProps,
+    {}
+> {
+    public static displayName: string = "ActionTrigger";
+
+    protected handledProps: HandledProps<ActionTriggerHandledProps> = {
+        appearance: void 0,
+        href: void 0,
+        managedClasses: void 0,
+        disabled: void 0,
+        glyph: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLButtonElement | HTMLAnchorElement> {
+        return (
+            <Button
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                disabled={this.props.disabled}
+                href={this.props.href}
+                appearance={
+                    ButtonAppearance[ActionTriggerAppearance[this.props.appearance]]
+                }
+                jssStyleSheet={actionTriggerButtonOverrides}
+            >
+                <div
+                    slot="before"
+                    className={get(this.props, "managedClasses.actionTrigger_glyph")}
+                >
+                    {this.props.glyph}
+                </div>
```

Comment:
Instead of a `<div>`, this should really be a `span`. However, I don't think it's necessary if we use a render prop. With a render prop, we can inform people that we will be passing a className to style the component; that removes the need for a generic outer wrapper like this. If they opt-in to the className, we can apply the class and set styles to whatever "thing" they pass without having to wrap this.

With a render prop, this might look like:
```
{ this.props.glyph(get(this.props, "managedClasses.actionTrigger_glyph")) }
```

Reply:
So I changed the wrapping element to a span but kept it in order to assign the glyph to the "before" slot.  Lumping it in the the children messed up the vertical alignment.  (and also if we aren't going to use slots we should be consistent between similar controls, like call to action in this case).

Reply:
How is this being lumped in with children if we are using a render prop? This is a net-new component and so the API should reflect the long-term API we would like to see. We are moving to render props; a user should not be concerned with the internals (we use the render prop and pass to another component "slot" property). 

It's a correct statement that we are moving away from slots, but in my opinion, we should not:

1. Use an API we intend to deprecate (slots)
2. Not do the right thing until we change all other component API's

Reply:
Got rid of slots, override button base styles to make content region a flex container so items can vertically align properly.